gui/src/components/Layout.tsx

```diff
@@ -218,54 +232,75 @@ const Layout = () => {
     };
   }, []);
 
+  useEffect(() => {
+    checkBump();
```

Comment:
Maybe just define checkBump as a nested function here to simplify and avoid the useCallback?